autoscheduler/scraper/views.py

```diff
@@ -1,4 +1,16 @@
-# Once we start adding routes, uncomment this line
-# from django.shortcuts import render
+from rest_framework import generics
+from .serializers import CourseSerializer
+from .models.course import Course
 
-# Create your views here.
+class RetrieveCourseView(generics.RetrieveAPIView):
+    """ API endpoint for viewing course information, used by /api/course.
+        This view returns a serialized course, should return its title and credit hours.
+    """
+    serializer_class = CourseSerializer
+
+    def get_object(self):
+        """ Overrides default behavior of get_object() to work without a primary key """

```

Comment:
What does you mean by without a primary key? Does `RetrieveApiView` just use the serializer's model and query using the parameters by default?

Reply:
By default, django rest framework requires you to provide the primary key of the item you're querying (see https://www.django-rest-framework.org/api-guide/generic-views/#get_objectself), and throws an error if it isn't provided, so I overwrote the `get_object(self)` method to instead get an object based on the query parameters.